stdlib/public/core/ArrayBuffer.swift

```diff
@@ -261,7 +261,7 @@ extension _ArrayBuffer {
 
       // Tell Cocoa to copy the objects into our storage
       cocoa.buffer.getObjects(
-        UnsafeMutablePointer(result.firstElementAddress),
```

Comment:
Seems like `getObjects` should accept an optional pointer, for the case when it is asked to copy zero objects.

Actually, you could add a fast path to the beginning of this function for that case.  We shouldn't need to do a message send for that.